### DISCUSSION TITLE
Can we able to merge All OAuth and email authenticated accounts?

### DISCUSSION THREAD

**Discussion by yahu1031** [serverpod/serverpod#576] [Q&A] [answered]
As of now, both apple and Google accounts will be able to merge automatically as I have read from docs/some issues. But can we also merge both OAuth accounts and email accounts? @vlidholt

**Comment by vlidholt** [comment, marked as answer]
Yes, this is already happening.

**Reply by yahu1031** [reply]
@vlidholt But how? If I have signed up the user with OAuth, Then user will not have any password, and if the user tries to log in, the only way is to use OAuth. Then how can we merge? By signing up again with a password?

**Reply by vlidholt** [reply]
Yes, if the emails match the accounts will be merged. You cannot create an email account without verifying the email address, so it's safe.

**Reply by yahu1031** [reply]
hmmmm, sounds true.
